ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor.hs

```diff
@@ -667,3 +702,110 @@ wakeupDecision st _now =
+         _ -> Unrestricted
+
+
+outboundConnectionsState
+    :: Ord peeraddr
+    => AssociationMode
+    -> PeerSelectionSetsWithSizes peeraddr
+    -> PeerSelectionState peeraddr peerconn
+    -> OutboundConnectionsState
+outboundConnectionsState
+    associationMode
+    PeerSelectionView {
+        viewEstablishedPeers = (viewEstablishedPeers, _),
+        viewEstablishedBootstrapPeers = (viewEstablishedBootstrapPeers, _),
+        viewActiveBootstrapPeers = (viewActiveBootstrapPeers, _)
+      }
+    PeerSelectionState { localRootPeers,
+                         publicRootPeers,
+                         bootstrapPeersFlag
+                       }
+    =
+    case (associationMode, bootstrapPeersFlag) of
+      {-
+       -- genesis mode
+       -- TODO: issue #4846
+      (LocalRootsOnly, _)
+                |  numberOfActiveBigLedgerPeers >= targetNumberOfActiveBigLedgerPeers
+                -> TrustedStateWithExternalPeers
+
+                |  otherwisej
+                -> UntrustedState
+      -}
+
+      (LocalRootsOnly, _)
+        |  -- we are only connected to trusted local root
+           -- peers or bootstrap peers
+           viewEstablishedPeers `Set.isSubsetOf` (viewEstablishedBootstrapPeers <> trustableLocalRootSet)
+        -> TrustedStateWithExternalPeers
+
+        | otherwise
+        -> UntrustedState
+
+       -- bootstrap mode
+      (Unrestricted, UseBootstrapPeers {})
+                |  -- we are only connected to trusted local root
+                   -- peers or bootstrap peers
+                   viewEstablishedPeers `Set.isSubsetOf` (viewEstablishedBootstrapPeers <> trustableLocalRootSet)
+                   -- there's at least one active bootstrap peer
+                ,  not (Set.null viewActiveBootstrapPeers)
+                -> TrustedStateWithExternalPeers
+
+                |  otherwise
+                -> UntrustedState
+
+       -- praos mode with public roots
+      (Unrestricted, DontUseBootstrapPeers)
+                | -- we are only connected to trusted local root
+                  -- peers and at least one public root peer
+                  --
+                  -- Since the node is promiscuous and don't use
+                  -- bootstrap peers, it must be using ledger peers.
+                  -- It's likely to have non-zero targets for big
+                  -- ledger peers or ledger peers, in which case the
+                  -- condition below likely will not be satisfied,
+                  -- and thus `UntrustedState` will be returned.
+                  let establishedPublicRootPeers =
+                        viewEstablishedPeers
+                        `Set.intersection`
+                        Map.keysSet (getPublicConfigPeers publicRootPeers)
+                , viewEstablishedPeers `Set.isSubsetOf` (establishedPublicRootPeers <> trustableLocalRootSet)
+                , not (null establishedPublicRootPeers)
+                -> TrustedStateWithExternalPeers
+
```

Comment:
Perhaps we should just simplify this and always return `Untrusted` state.
The node is not configured to use bootstrap peers and while it is possible that it may be in a trusted state out of pure luck I don't think it is worth the effort to check.

This could also confuse operators when we have exposed the counters for this. The operator thinks he has disabled bootstrap peers but once in a blue moon it is active anyway.

Reply:
Yes, that's fine.  We really need to handle just the genesis and bootstrap modes, the praos mode can stay unchanged.